public/app/features/alerting/unified/utils/rules.ts

```diff
@@ -47,49 +49,72 @@ import { GRAFANA_ORIGIN_LABEL } from './labels';
+function isGrafanaAlertingRule(rule?: RulerRuleDTO): rule is RulerGrafanaRuleDTO<GrafanaAlertingRuleDefinition> {
+  return isGrafanaRulerRule(rule) && !isGrafanaRecordingRule(rule);
 }
 
-export function isRecordingRulerRule(rule?: RulerRuleDTO): rule is RulerRecordingRuleDTO {
-  return typeof rule === 'object' && 'record' in rule;
+function isGrafanaRecordingRule(rule?: RulerRuleDTO): rule is RulerGrafanaRuleDTO<GrafanaRecordingRuleDefinition> {
+  return isGrafanaRulerRule(rule) && 'record' in rule.grafana_alert;
 }
 
-export function isGrafanaOrDataSourceRecordingRule(rule?: RulerRuleDTO) {
-  return (
-    (typeof rule === 'object' && isRecordingRulerRule(rule)) ||
-    (isGrafanaRulerRule(rule) && 'record' in rule.grafana_alert)
-  );
+function isGrafanaRulerRulePaused(rule?: RulerRuleDTO) {
+  return isGrafanaRulerRule(rule) && Boolean(rule.grafana_alert.is_paused);
 }
 
-export function isGrafanaRecordingRule(rule?: RulerRuleDTO): rule is RulerGrafanaRuleDTO {
-  return typeof rule === 'object' && isGrafanaOrDataSourceRecordingRule(rule) && isGrafanaRulerRule(rule);
+/* Data source managed rules */
+
+function isAlertingRulerRule(rule?: RulerRuleDTO): rule is RulerAlertingRuleDTO {
+  return typeof rule === 'object' && 'alert' in rule;
 }
 
-export function isGrafanaAlertingRule(rule?: RulerRuleDTO): rule is RulerGrafanaRuleDTO {
-  return typeof rule === 'object' && isGrafanaRulerRule(rule) && !isGrafanaOrDataSourceRecordingRule(rule);
+function isCloudRulerRule(rule?: RulerRuleDTO | PostableRuleDTO): rule is RulerCloudRuleDTO {
+  return typeof rule === 'object' && !isGrafanaRulerRule(rule);
 }
 
-export function isGrafanaRulerRule(rule?: RulerRuleDTO | PostableRuleDTO): rule is RulerGrafanaRuleDTO {
-  return typeof rule === 'object' && 'grafana_alert' in rule;
+function isCloudRecordingRulerRule(rule?: RulerRuleDTO): rule is RulerRecordingRuleDTO {
+  return typeof rule === 'object' && 'record' in rule;
 }
 
-export function isCloudRulerRule(rule?: RulerRuleDTO | PostableRuleDTO): rule is RulerCloudRuleDTO {
-  return typeof rule === 'object' && !isGrafanaRulerRule(rule);
+/* Prometheus rules */
+
+function isAlertingRule(rule?: Rule): rule is AlertingRule {
+  return typeof rule === 'object' && rule.type === PromRuleType.Alerting;
 }
 
-export function isGrafanaRulerRulePaused(rule: RulerGrafanaRuleDTO) {
-  return rule && isGrafanaRulerRule(rule) && Boolean(rule.grafana_alert.is_paused);
+function isRecordingRule(rule?: Rule): rule is RecordingRule {
+  return typeof rule === 'object' && rule.type === PromRuleType.Recording;
 }
 
+export const rulerRuleType = {
+  grafanaManaged: {
+    rule: isGrafanaRulerRule,
+    alertingRule: isGrafanaAlertingRule,
+    recordingRule: isGrafanaRecordingRule,
+    pausedRule: isGrafanaRulerRulePaused,
+  },
+  dataSourceManaged: {
+    rule: isCloudRulerRule,
+    alertingRule: isAlertingRulerRule,
+    recordingRule: isCloudRecordingRulerRule,
```

Comment:
nit: How about 
- `grafanaManaged` -> `grafana`, 
- `dataSourceManaged` -> `dataSource`
Just to make it a bit shorter 😄